### ISSUE TITLE
parameter 'padding' of torch.nn.MaxPool1d can be set to negative integral

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#83224]
### 🐛 Describe the bug

Parameter 'padding' of torch.nn.MaxPool1d must be >= 0 and <= kernel_size / 2 in documentation. However, I found that negative integral can be set to torch.nn.MaxPool2d.

```
import torch
results={}
arg_1 = 2
arg_2 = 50
arg_3 = True
arg_class = torch.nn.MaxPool1d(arg_1,padding=-1,stride=arg_2,return_indices=arg_3,)
arg_4 = torch.rand([0, 1, 49], dtype=torch.float32)
results['res'] = arg_class(arg_4)
```
Above code works.

### Versions

pytorch: 1.8.1
python: 3.8.3
os: win11

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by cheyennee** [comment]
But if padding is set to negative integral, torch.nn.functional.max_pool1d will throw runtimeerror message. #max_pool1d() padding must be non-negative, but got -1

```
import torch
results={}
arg_1_tensor = torch.rand([1, 1, 9], dtype=torch.float32)
arg_1 = arg_1_tensor.clone()
arg_2 = 2
arg_3 = 2
arg_4 = -1
arg_5 = 1
arg_6 = False
arg_7 = False
results['res'] = torch.nn.functional.max_pool1d(arg_1,arg_2,arg_3,arg_4,arg_5,arg_6,arg_7,)
```
